src/System.IO.FileSystem/src/System/IO/Win32FileStream.cs

```diff
@@ -1118,28 +1113,7 @@ private unsafe void WriteCore(byte[] buffer, int offset, int count)
 
 #if USE_OVERLAPPED
         [System.Security.SecuritySafeCritical]  // auto-generated
-        internal IAsyncResult BeginRead(byte[] array, int offset, int numBytes, AsyncCallback userCallback, Object stateObject)
-        {
-            if (array == null)
-                throw new ArgumentNullException("array");
-            if (offset < 0)
-                throw new ArgumentOutOfRangeException("offset", SR.ArgumentOutOfRange_NeedNonNegNum);
-            if (numBytes < 0)
-                throw new ArgumentOutOfRangeException("numBytes", SR.ArgumentOutOfRange_NeedNonNegNum);
-            if (array.Length - offset < numBytes)
-                throw new ArgumentException(SR.Argument_InvalidOffLen);
-            Contract.EndContractBlock();
-
-            if (_handle.IsClosed) throw __Error.GetFileNotOpen();
-
-            if (!_isAsync)
-                return _streamAsyncHelper.BeginRead(array, offset, numBytes, userCallback, stateObject);
-            else
-                return BeginReadAsync(array, offset, numBytes, userCallback, stateObject);
-        }
-
-        [System.Security.SecuritySafeCritical]  // auto-generated
-        private FileStreamAsyncResult BeginReadAsync(byte[] array, int offset, int numBytes, AsyncCallback userCallback, Object stateObject)
+        private Task<int> BeginReadAsync(byte[] array, int offset, int numBytes, CancellationToken cancellationToken)
```

Comment:
No longer need the "Begin" prefix.  This could be ReadAsyncCore or something like that, but "Begin" is usually reserved for when there's a corresponding "End".